python/across_api/base/fov.py

```diff
@@ -0,0 +1,354 @@
+        healpix_nside
+                The NSIDE value of the HEALPix map. If not provided, it will be
+                calculated based on the length of healpix_loc.
+        healpix_order
+                The ordering scheme of the HEALPix map. Default is "NESTED".
+
+        Returns
+        -------
+        float
+                The amount of probability inside the FOV.
+
+        """
+        # Extract the NSIDE value from the HEALPix map, also level and ipix if
+        # this is a MOC map
+        if healpix_order == "NUNIQ":
+            level, ipix = ah.uniq_to_level_ipix(healpix_loc["UNIQ"])
+            uniq_nside = ah.level_to_nside(level)
+            healpix_loc = healpix_loc["PROBDENSITY"]
+        else:
+            nside = ah.npix_to_nside(len(healpix_loc))
+
+        # Find where in HEALPix map the probability is > 0
+        nonzero_prob_pixels = np.where(healpix_loc > 0.0)[0]
+
+        # Create a list of RA/Dec coordinates for these pixels
+        if healpix_order == "NUNIQ":
+            ra, dec = ah.healpix_to_lonlat(
+                ipix[nonzero_prob_pixels],
+                nside=uniq_nside[nonzero_prob_pixels],  # type: ignore
+                order="NESTED",
+            )
+        else:
+            ra, dec = ah.healpix_to_lonlat(
+                nonzero_prob_pixels, nside=nside, order=healpix_order
+            )
+
+        # Convert these coordinates into a SkyCoord
+        skycoord = SkyCoord(ra=ra, dec=dec, unit="deg")
+
+        # Calculate pixel indicies of the all the regions inside of the FOV
+        visible_probability_pixels = nonzero_prob_pixels[self.in_fov(skycoord=skycoord)]
+        # Calculate the amount of probability inside the FOV
+        if healpix_order == "NUNIQ":
+            # Calculate probability in FOV by multiplying the probability density by
+            # area of each pixel and summing up
+            pixarea = ah.nside_to_pixel_area(uniq_nside[visible_probability_pixels])
+            return np.sum(healpix_loc[visible_probability_pixels] * pixarea.value)
+        else:
+            # Calculate the amount of probability inside the FOV
+            return np.sum(healpix_loc[visible_probability_pixels])
+
+
+class FootprintFOV(FOVBase):
+    """
+    Constrained instrumet FOV. This is an FOV that calculate lates what is
+    visible in the FOV at a given pointing
+    """
+
+    footprint: Footprint
+
+    def __init__(self, pointing: PointingBase) -> None:
+        center = SkyCoord(pointing.ra, pointing.dec, unit="deg")
+
+        projected_footprint = self.footprint.project(
+            center=center, pos_angle=pointing.position_angle * u.deg
+        )
+
+        ras_radians = projected_footprint.ra.to(u.rad).value[:-1]
+        decs_radians = projected_footprint.dec.to(u.rad).value[:-1]
+
+        cartesian_vertices = spherical_to_cartesian(
+            1,  # radial component
+            decs_radians,  # LAT - DECs
+            ras_radians,  # LON - RAs
+        )
```

Comment:
```suggestion
        cartesian_vertices = projected_footprint.cartesian.xyz.value
```

I'm not sure about the shape; you might need to remove the transpose below.

Reply:
This works, no other changes needed: 

`cartesian_vertices = projected_footprint[:-1].cartesian.xyz.value`